### PR TITLE
teeworlds: cmake instead of bam, oh joy! cleanup

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

Looks like teeworlds offers cmake support
and since it is mentioned first in their README
I gave it a go and ... :grin: :tada:

This fixes long-standing issues I've had building teeworlds
with more exotic compilers
(including clang? but havne't tried it using nixpkgs-master lately...)
and makes it possible to greatly simplify the expression
with all the same files installed AFAICT.

I noticed bam now has no consumers in-tree,
but I'm content letting it sit there (and maybe bitrot if untested)
as long as it doesn't make things too awkward when we're hanging
and it sees teeworlds expression happier and healthier than ever before
and while that truly brings joy it hurts and is a constant reminder of
its mistakes and weaknesses and what could have been ....
well so I think we can maybe hold off on evicting it for now.
That seems kinda cruel.  Likely teeworlds wouldn't be what it is today
if not for all bam has done for it.

Okay anyway >_> <_<.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I've only tested the client?
And now that there's a build system we're familiar with... if anyone's
curious we might have some options to poke at.

(sorry bam ;3)

### CODE REVIEW

**Review by tomberek — APPROVED** [review]
Builds and runs on NixOS.

Love the excitement in the PR.

File: pkgs/games/teeworlds/default.nix [hunk 1]
```diff
@@ -21,35 +21,12 @@ stdenv.mkDerivation rec {
...
-  '';
-
-  installPhase = ''
-    mkdir -p $out/bin $out/share/teeworlds
-    cp build/x86_64/release/teeworlds{,_srv} $out/bin
-    cp -r build/x86_64/release/data $out/share/teeworlds
-  '';
-
-  buildInputs = [
+    buildInputs = [
```
Comment (Mic92) [inline pkgs/games/teeworlds/default.nix]:
```suggestion
  buildInputs = [
```

Comment (dtzWill) [reply pkgs/games/teeworlds/default.nix]: Haha, d'oh! But thank you for catching and bringing it to my attention.  Fixup'd it into the main commit, should be good now :).

**Comment by prusnak** [comment]
Can we just change `platforms` and `homepage` while we are at it?

```
platforms = stdenv.lib.platforms.linux;
homepage = "https://teeworlds.com/";
```

**Comment by ajs124** [comment]
Has anything ever depended on `bam`? I seem to remember it as "the teeworlds build system".

**Comment by Mic92** [comment]
In that case `bam` could be even dropped.

**Comment by dtzWill** [comment]
Will leave dropping bam to a follow-up PR, :innocent: .

**Pull request merged**